spec/rubocop/cop/rspec/repeated_example_group_body_spec.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::RSpec::RepeatedExampleGroupBody do
+  subject(:cop) { described_class.new }
+
+  it 'registers an offense for repeated describe body' do
+    expect_offense(<<-RUBY)
+      describe 'doing x' do
+      ^^^^^^^^^^^^^^^^^^^^^ Repeated describe block body on line(s) [5]
+        it { cool_predicate_method }
+      end
+
+      describe 'doing y' do
+      ^^^^^^^^^^^^^^^^^^^^^ Repeated describe block body on line(s) [1]
+        it { cool_predicate_method }
+      end
+    RUBY
+  end
+
+  it 'registers an offense for repeated context body' do
+    expect_offense(<<-RUBY)
+      context 'when awesome case' do
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Repeated context block body on line(s) [5]
+        it { cool_predicate_method }
+      end
+
+      context 'when another awesome case' do
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Repeated context block body on line(s) [1]
+        it { cool_predicate_method }
+      end
+    RUBY
+  end
+
+  it 'does not register offense for different block body implementation' do
+    expect_no_offenses(<<-RUBY)
+      context 'when awesome case' do
+        it { cool_predicate_method }
+      end
+
+      context 'when another awesome case' do
+        it { another_predicate_method }
+      end
+    RUBY
+  end
+
+  it 'does not register offense if rspec tag magic is involved' do
+    expect_no_offenses(<<-RUBY)
+      describe 'doing x' do
+        it { cool_predicate_method }
+      end
+
+      describe 'doing x', :request do
+        it { cool_predicate_method }
+      end
+    RUBY
+  end
+
+  it 'does not register offense for different groups' do
+    expect_no_offenses(<<-RUBY)
+      describe 'doing x' do
+        it { cool_predicate_method }
+      end
+
+      context 'doing x' do
+        it { cool_predicate_method }
+      end
+    RUBY
+  end
```

Comment:
Another thing that is nice to be certain about is that it only flags groups that are in the exact same group only, e.g.
```
describe 'A' do
  describe '.b' do
    context 'when this' do
    end
  end
  describe '.c' do
    context 'when this' do
    end
  end
  context 'when this' do
  end
end
```
Nothing here should be flagged.

Reply:
@pirj it should flag `describe .b and .c` because they have a similar body. Maybe the main point here that it should not flag `context` in and out of `describe` group`?

Reply:
Sorry for the confusion, edited the comment. I tried to empasize on `context 'when this'`.